src/mds/SessionMap.h

```diff
@@ -344,6 +349,11 @@ class Session : public RefCountedObject {
     }
   }
 
+  void set_connection(Connection *con) {
```

Comment:
There isn't any enforcement that somebody uses set_connection() instead of just grabbing the ConnectionRef directly — if it's important and we can't reliably force them by marking the member private, we'd better at least comment that direct setting isn't allowed...